src/ray/raylet/actor_registration.h

```diff
@@ -118,8 +132,10 @@ class ActorRegistration {
   /// dependencies. This is indexed by handle.
   std::unordered_map<ActorHandleID, FrontierLeaf> frontier_;
 
-  /// All of the dummy object IDs from this actor's tasks.
-  std::vector<ObjectID> dummy_objects_;
+  /// A map from all of the dummy object IDs stored for this
+  /// actor to the number of actor handles that have the object
+  /// ID in scope.
+  std::unordered_map<ObjectID, int64_t> dummy_objects_;
```

Comment:
I think this field would be a good place to provide a fairly in-depth description of how we determine precisely which dummy objects must be kept around and when we know that one can be marked as unavailable.

Someone trying to understand this code will probably look here.